File: python/bob/bob.py
def response(hey_bob):
    hey_bob = hey_bob.rstrip().lstrip()
    is_question = len(hey_bob) == 0 or hey_bob[-1] == '?'
    is_empty = len([c for c in hey_bob if c.isspace()]) == len(hey_bob)
    has_words = len([c for c in hey_bob if c.isalpha()]) == 0
    is_yelling = hey_bob == hey_bob.upper() and not has_words

    if is_empty:
        response = "Fine. Be that way!"
    elif is_yelling and is_question:
        response = "Calm down, I know what I'm doing!"
    elif is_question:
        response = "Sure."
    elif is_yelling:
        response  = "Whoa, chill out!"
    else:
        response = "Whatever."

    return response

File: java/bob/src/main/java/Bob.java
class Bob {

    String hey(String input) {
        var letters = input.chars().filter(Character::isAlphabetic).count();
        var isYelling = input.chars().filter(Character::isAlphabetic).filter(Character::isUpperCase).count() ==letters && letters > 0;
        var isQuestion = input.trim().endsWith("?");
        var response = "Whatever.";

        /*
        "Sure." This is his response if you ask him a question, such as "How are you?" The convention used for questions is that it ends with a question mark.
"Whoa, chill out!" This is his answer if you YELL AT HIM. The convention used for yelling is ALL CAPITAL LETTERS.
"Calm down, I know what I'm doing!" This is what he says if you yell a question at him.
"Fine. Be that way!" This is how he responds to silence. The convention used for silence is nothing, or various combinations of whitespace characters.
"Whatever." This is what he answers to anything else.
         */


        if( isYelling && isQuestion){
            response = "Calm down, I know what I'm doing!";
        }
        else if(isQuestion){
            response = "Sure.";
        }
        else if(isYelling){
            response = "Whoa, chill out!";
        }
        else if(input.trim().isEmpty()) {
            response = "Fine. Be that way!";
        }

        return response;
    }

}